homeassistant/components/sensor/thermoworks_smoke.py

```diff
@@ -0,0 +1,161 @@
+    vol.Optional(CONF_EXCLUDE, default=[]):
+        vol.All(cv.ensure_list),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the thermoworks sensor."""
+    import thermoworks_smoke
+    from requests.exceptions import HTTPError
+
+    email = config.get(CONF_EMAIL)
+    password = config.get(CONF_PASSWORD)
+    monitored_variables = config.get(CONF_MONITORED_VARIABLES)
+    excluded = config.get(CONF_EXCLUDE)
+
+    try:
+        mgr = thermoworks_smoke.initialize_app(email, password, True, excluded)
+
+        # list of sensor devices
+        dev = []
+
+        # get list of registered devices
+        for serial in mgr.serials():
+            for variable in monitored_variables:
+                dev.append(ThermoworksSmokeSensor(variable, serial, mgr))
+
+        add_devices(dev, True)
+    except HTTPError as error:
+        msg = "{}".format(error.strerror)
+        if 'EMAIL_NOT_FOUND' in msg or \
+                'INVALID_PASSWORD' in msg:
+            _LOGGER.error("Invalid email and password combination")
+        else:
+            _LOGGER.error(msg)
+
+
+class ThermoworksSmokeSensor(Entity):
+    """Implementation of a thermoworks smoke sensor."""
+
+    def __init__(self, sensor_type, serial, mgr):
+        """Initialize the sensor."""
+        self._name = "{name} {sensor}".format(name=mgr.name(serial),
+                                              sensor=SENSOR_TYPES[sensor_type])
+        self.type = sensor_type
+        self._state = None
+        self._attributes = {}
+        self._unit_of_measurement = TEMP_FAHRENHEIT
+        self.serial = serial
+        self.mgr = mgr
+        self.update_unit()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._attributes
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this sensor."""
+        return self._unit_of_measurement
+
+    def update_unit(self):
+        """Set the units from the data."""
+        if PROBE_2 in self.type:
+            self._unit_of_measurement = self.mgr.units(self.serial, PROBE_2)
+        else:
+            self._unit_of_measurement = self.mgr.units(self.serial, PROBE_1)
+
+    def update(self):
+        """Get the monitored data from firebase."""
+        try:
+            values = self.mgr.data(self.serial)
+
+            # set state from data based on type of sensor
+            self._state = values.get(self.type)
+
+            # set units
+            self.update_unit()
+
+            # set basic attributes for all sensors
+            self._attributes = {
+                'serial': self.serial,
+                'time': values['time'],
```

Comment:
What kind of time is this? We are not allowed to store time in state attributes that changes on each update, just because time has passed, ie not because state has changed.

Reply:
No, this is the time new data was pushed by the thermoworks probe. It's telling you it was that temperature at that time.